### PR TITLE
Feature/5460 oauth webflow

### PR DESCRIPTION
**Pull request by johannespfeiffer**
This PR explains how to use the new OAuth2 endpoints and removes the description of the deprecated JWT/Basic Auth endpoints.

See https://app.zenhub.com/workspaces/kontist-engineering-5be06ab44b5806bc2bf14eca/issues/figureaps/figure-backend/5579

### CODE REVIEW

File: source/index.html.md [hunk 1]
```diff
@@ -18,91 +18,121 @@ search: true
...
 
 # Authentication
+To manage the data via our API your application needs to gain access on behalf of the user. This is done through obtaining an access token via [OAuth2](https://tools.ietf.org/html/rfc6749). The access token must then be send in each request in the HTTP header like this: "Authorization: Bearer TOKEN".
 
-You can start playing with the Kontist API using [HTTP Basic Auth](https://en.wikipedia.org/wiki/Basic_access_authentication). However this is not to be used in production as it is highly rate limited. The documentation of our oAuth flow is in preparation and will be added at a later point in time.
+You need two kinds of credentials to get such a token: The first part is a fixed pair of client id and client secret. They identify your client application which connects to the API. Each application has its own pair of client id and secret, please contact us via developer@kontist.com to get your own client credentials.
```
Comment (hihuz) [inline source/index.html.md:23]: this email address actually exists? 🙂Do you know who it will send the email to?

File: source/index.html.md [hunk 2]
```diff
@@ -18,91 +18,121 @@ search: true
...
+The second part is obtained through the user and can be done in several ways, here we describe the preferred way through the "Authorization Code" grant type.
 
-`Authorization: Basic QWxhZGRpbjpPcGVuU2VzYW1l`
+## Authorization Code
+In general, the process looks like this:
+1. You redirect the user in a browser to an url on our end.
+2. The user is required to login and needs to accept your applications authorization request. The browser redirects back to your application with a `
code` parameter.
```
Comment (hihuz) [inline source/index.html.md]: Nit: Shouldn't it be `your application's authorization request` with a `'` ?

File: source/index.html.md [hunk 3]
```diff
@@ -18,91 +18,121 @@ search: true
...
 
-You can read more about JWT token auth under [https://jwt.io/introduction](https://jwt.io/introduction).
+Adjust the parameters like this:
+| Parameter | Description |
+| --------- | ----------- |
+| scope | space delimited list of scopes your application is going to access. Please see the list below.|
+| response_type | Set fixed as "code". |
+| client_id | This is your client id you got from us. Do not include the secret here.|
+| redirect_uri | This is your applications callback url which is bound to your client id.|
+| state | Can be used to verify our response. You can put in anything here and we will send it back to your application. |later.
```
Comment (hihuz) [inline source/index.html.md]: what's with this `later.` at the end after closing the table? 😄is it on purpose?

File: source/index.html.md [hunk 4]
```diff
@@ -18,91 +18,121 @@ search: true
...
+```
 
-You can read more about JWT token auth under [https://jwt.io/introduction](https://jwt.io/introduction).
+Adjust the parameters like this:
+| Parameter | Description |
+| --------- | ----------- |
+| scope | space delimited list of scopes your application is going to access. Please see the list below.|
+| response_type | Set fixed as "code". |
+| client_id | This is your client id you got from us. Do not include the secret here.|
+| redirect_uri | This is your applications callback url which is bound to your client id.|
```
Comment (hihuz) [inline source/index.html.md]: same nit here, maybe we could write `application's`?

**Review by hihuz — APPROVED** [review]
Looks great 👍

Left a couple very minor comments

File: source/index.html.md (hunk 1 above)
Comment (johannespfeiffer) [reply source/index.html.md:23]: Yes, it is a redirect and I confirmed that it actually works ;-)

File: source/index.html.md (hunk 2 above)
Comment (johannespfeiffer) [reply source/index.html.md]: Sure, done.

File: source/index.html.md (hunk 4 above)
Comment (johannespfeiffer) [reply source/index.html.md]: Sure, done.

File: source/index.html.md (hunk 3 above)
Comment (johannespfeiffer) [reply source/index.html.md]: Done. ;-)

**Review by hihuz — APPROVED** [review]
💯

**Comment by johannespfeiffer** [comment]
Let us wait with the merge until https://github.com/figureaps/figure-backend/pull/5600 has been deployed.

**Pull request merged**